backend/user_service/src/usermanager.js

```diff
@@ -129,27 +130,43 @@ const getUserById = (request, response) => {
     })};
 
 const updateUser = (request, response) => {
-    const name = request.body.name;
-    const userId = parseInt(request.params.id);
-    const query = `UPDATE users SET name = $1 WHERE id = $2 RETURNING id`;
 
-    if (!request.body || !name || name.length == 0) {
-        // Reject if 'name' is empty;
-        const msg = {'msg': `Name cannot be empty.`, 'id': null};
+    const {name, token} = request.body;
+    const userId = parseInt(request.params.id);
+    if (!request.body || (!name || name.length == 0) && (!token || token.length == 0)) {
+        // Reject if all fields are empty;
+        const msg = {'msg': `At least one field must be filled.`};
         return response.status(400).json(msg);
     }
-    if (name.length > 255) {
+    if (name && name.length > 255) {
```

Comment:
Good job on catching the constriant. Perhaps might want to reduce the length of the name to a smaller length. 

From the frontend perspective, 255 characters might be too long and might affect the visibilty of the table.

Reply:
This constraint is to fit into PSQL's VARCHAR(255) length limit. Additional limits can be done on the API gateway level.